tests/reports/class-wc-tests-reports-orders-stats.php

```diff
@@ -3287,7 +3287,7 @@ public function test_segmenting_by_product_and_variation() {
 		$customer_1 = WC_Helper_Customer::create_customer( 'cust_1', 'pwd_1', 'user_1@mail.com' );
 
 		$order_1_time = time();
-		$order_3_time = $order_1_time - 1 * HOUR_IN_SECONDS;
+		$order_3_time = $order_1_time - 1 * HOUR_IN_SECONDS; // @todo: I wonder if this messes up new/returning customer
```

Comment:
Very good question.  It's okay right now, but there are 2 potential places where this would create issues:

1.  In [this issue](https://github.com/woocommerce/wc-admin/issues/1490#issuecomment-460917176), I proposed a potential solution that tests against IDs as well as dates.  Since `$order_3` would have a larger ID, this would fail.  

I think the solution I proposed there is a bad one though, so this is probably a non-issue.

2.  In this case, because `$order_1` is created first, but `$order_3` has an earlier date, `$order_1` should be marked as returning.  I touched on this at the end of [this issue](https://github.com/woocommerce/wc-admin/issues/1473):

>Furthermore, we would also need to check all orders by a customer and update if a status is changed on a previous order.

If an order date is updated, we would need to update all their orders again with the correct returning value if the order date is earlier than all previous orders.

Reply:
This should be a non-issue after https://github.com/woocommerce/wc-admin/pull/1517 is merged in.